client/core/core.go

```diff
@@ -1238,11 +1259,13 @@ func (c *Core) CreateWallet(appPW, walletPW []byte, form *WalletForm) error {
 func (c *Core) loadWallet(dbWallet *db.Wallet) (*xcWallet, error) {
 	c.connMtx.RLock() // required to calculate contractlocked amount
 	defer c.connMtx.RUnlock()
+	contractLockedAmt, orderLockedAmt := c.lockedAmounts(dbWallet.AssetID)
```

Comment:
Looking at the two uses of `lockedAmounts`, I think you should move the lock inside now.